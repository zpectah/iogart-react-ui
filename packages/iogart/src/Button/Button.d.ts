import { ButtonProps } from './types';

declare const Button: ButtonProps;

export default Button;
