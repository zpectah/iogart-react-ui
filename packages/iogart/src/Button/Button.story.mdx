import { Meta, Story } from '@storybook/addon-docs';

import Button from './Button';
export const Template = (args) => <Button {...args} />;

<Meta title="Components/Action/Button" component={Button} />

# Button

`@iogart-react-ui/iogart/Button`

```javascript
import { Button } from 'iogart-react-ui/iogart';
```

<Story
  name="Default"
  args={{
    children: 'Default Button',
  }}
>
  {Template.bind({})}
</Story>

<Story
  name="Primary"
  args={{
    children: 'Primary Button',
    // onDoubleClick: (e) => { console.log('onDoubleClick e', e) },
  }}
>
  {Template.bind({})}
</Story>
