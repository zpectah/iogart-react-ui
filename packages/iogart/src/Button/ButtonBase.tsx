import React, { createElement } from 'react';

import { ButtonBaseProps } from './types';
import useButtonBase from './useButtonBase';

const ButtonBase = (props: ButtonBaseProps) => {
  const { elementType = 'button', children, as, ...rest } = props;

  const Element = as ? as : elementType;
  const updatedProps = useButtonBase({ ...rest });

  return createElement(Element, updatedProps, children);
};

export default ButtonBase;
