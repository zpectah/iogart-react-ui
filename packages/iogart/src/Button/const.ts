export const BUTTON_BASE_CLASSNAME = 'ButtonBase';
export const BUTTON_CLASSNAME = 'Button';
