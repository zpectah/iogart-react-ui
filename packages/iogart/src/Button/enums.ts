export enum ButtonElementTypeKeys {
  button = 'button',
  a = 'a',
}

export enum ButtonVariantKeys {
  contained = 'contained',
  outlined = 'outlined',
  link = 'link',
}

export enum ButtonSizeKeys {
  small = 'small',
  medium = 'medium',
  large = 'large',
}

export enum ButtonColorKeys {
  primary = 'primary',
  secondary = 'secondary',
  success = 'success',
  error = 'error',
  info = 'info',
  warning = 'warning',
}
