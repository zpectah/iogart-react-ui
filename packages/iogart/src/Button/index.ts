export { default as Button } from './Button';
export { default as useButton } from './useButton';
export * from './types';
export * from './Button.d';

