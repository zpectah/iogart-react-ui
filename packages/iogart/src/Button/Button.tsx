import React, { createElement } from 'react';

import { ButtonProps } from './types';
import ButtonBase from './ButtonBase';
import useButton from './useButton';

const Button = (props: ButtonProps) => {
  const { children, ...rest } = props;

  const updatedProps = useButton({ ...rest });

  return createElement(ButtonBase, updatedProps, children);
};

export default Button;
