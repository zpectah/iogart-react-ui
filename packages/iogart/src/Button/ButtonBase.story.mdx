import { Meta, Story } from '@storybook/addon-docs';

import ButtonBase from './ButtonBase';
export const Template = (args) => <ButtonBase {...args} />;

<Meta title="Components/Action/Button" component={ButtonBase} />

# Button Base

`@iogart-react-ui/base/ButtonBase`

```javascript
import { ButtonBase } from 'iogart-react-ui/base';
```

<Story
  name="Base"
  args={{
    children: 'Button Base',
  }}
>
  {Template.bind({})}
</Story>
