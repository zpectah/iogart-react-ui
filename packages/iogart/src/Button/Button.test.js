import renderer from 'react-test-renderer';

import Button from './Button';

it('Button - onMouseEnter & onMouseLeave', () => {

    const component = renderer.create(
        <Button>Button</Button>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
        tree.props?.onMouseEnter();
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    renderer.act(() => {
       tree.props?.onMouseLeave();
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
