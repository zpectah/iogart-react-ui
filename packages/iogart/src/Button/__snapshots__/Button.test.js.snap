// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button - onMouseEnter & onMouseLeave 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1jnqqvb{margin:0;padding:.5rem 1rem;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}",
      }
    }
    data-emotion="css 1jnqqvb"
  />,
  <button
    className="iogart-ui iogart-base iogart-button-base iogart-button css-1jnqqvb"
    onBlur={[Function]}
    onClick={[Function]}
    onDoubleClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={{}}
    type="button"
  >
    Button
  </button>,
]
`;

exports[`Button - onMouseEnter & onMouseLeave 2`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1jnqqvb{margin:0;padding:.5rem 1rem;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}",
      }
    }
    data-emotion="css 1jnqqvb"
  />,
  <button
    className="iogart-ui iogart-base iogart-button-base iogart-button css-1jnqqvb"
    onBlur={[Function]}
    onClick={[Function]}
    onDoubleClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={{}}
    type="button"
  >
    Button
  </button>,
]
`;

exports[`Button - onMouseEnter & onMouseLeave 3`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1jnqqvb{margin:0;padding:.5rem 1rem;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;}",
      }
    }
    data-emotion="css 1jnqqvb"
  />,
  <button
    className="iogart-ui iogart-base iogart-button-base iogart-button css-1jnqqvb"
    onBlur={[Function]}
    onClick={[Function]}
    onDoubleClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={{}}
    type="button"
  >
    Button
  </button>,
]
`;
